### ISSUE TITLE
yolov8 serialization error on custom dataset

### ISSUE THREAD

**Issue by LeRoii** [wang-xinyu/tensorrtx#1362]
## Env

- GPU, NVIDIA TITAN Xp
- OS, Ubuntu20.04
- Cuda version 11.3
- TensorRT version 8.2.4

## About this repo

- which branch/tag/commit are you using?  3cbb7cd25796509cb49cc2251020d589d98228c9
- which model?  yolov8
## Your problem
I trained yolov8 on Visdrone 2019. and generated engine file with gen_wts.py,
and I changed kNumClass in config.h, when I ran
 ` ./yolov8 -s visdrone-8n-10c.wts visdrone-8n-10c.engine n`
I got error output
```
Loading weights: /space/model/visdrone-8n-10c.wts
[09/11/2023-23:44:03] [W] [TRT] The implicit batch dimension mode has been deprecated. Please create the network with NetworkDefinitionCreationFlag::kEXPLICIT_BATCH flag whenever possible.
[09/11/2023-23:44:03] [E] [TRT] 3: (Unnamed Layer* 239) [Convolution]:kernel weights has count 36864 but 46080 was expected
[09/11/2023-23:44:03] [E] [TRT] 4: (Unnamed Layer* 239) [Convolution]: count of 36864 weights in kernel, but kernel dimensions (3,3) with 64 input channels, 80 output channels and 1 groups were specified. Expected Weights count is 64 * 3*3 * 80 / 1 = 46080
[09/11/2023-23:44:03] [E] [TRT] 4: [convolutionNode.cpp::computeOutputExtents::43] Error Code 4: Internal Error ((Unnamed Layer* 239) [Convolution]: number of kernel weights does not match tensor dimensions)
[09/11/2023-23:44:03] [E] [TRT] 3: (Unnamed Layer* 239) [Convolution]:kernel weights has count 36864 but 46080 was expected
[09/11/2023-23:44:03] [E] [TRT] 4: (Unnamed Layer* 239) [Convolution]: count of 36864 weights in kernel, but kernel dimensions (3,3) with 64 input channels, 80 output channels and 1 groups were specified. Expected Weights count is 64 * 3*3 * 80 / 1 = 46080
[09/11/2023-23:44:03] [E] [TRT] 4: [convolutionNode.cpp::computeOutputExtents::43] Error Code 4: Internal Error ((Unnamed Layer* 239) [Convolution]: number of kernel weights does not match tensor dimensions)
[09/11/2023-23:44:03] [E] [TRT] 3: (Unnamed Layer* 239) [Convolution]:kernel weights has count 36864 but 46080 was expected
[09/11/2023-23:44:03] [E] [TRT] 4: (Unnamed Layer* 239) [Convolution]: count of 36864 weights in kernel, but kernel dimensions (3,3) with 64 input channels, 80 output channels and 1 groups were specified. Expected Weights count is 64 * 3*3 * 80 / 1 = 46080
[09/11/2023-23:44:03] [E] [TRT] 4: [convolutionNode.cpp::computeOutputExtents::43] Error Code 4: Internal Error ((Unnamed Layer* 239) [Convolution]: number of kernel weights does not match tensor dimensions)
[09/11/2023-23:44:03] [E] [TRT] 3: (Unnamed Layer* 239) [Convolution]:kernel weights has count 36864 but 46080 was expected
[09/11/2023-23:44:03] [E] [TRT] 4: (Unnamed Layer* 239) [Convolution]: count of 36864 weights in kernel, but kernel dimensions (3,3) with 64 input channels, 80 output channels and 1 groups were specified. Expected Weights count is 64 * 3*3 * 80 / 1 = 46080
[09/11/2023-23:44:03] [E] [TRT] 4: [convolutionNode.cpp::computeOutputExtents::43] Error Code 4: Internal Error ((Unnamed Layer* 239) [Convolution]: number of kernel weights does not match tensor dimensions)
```

**Comment by LeRoii** [comment]
it seems that kNumClass in config.h didn't work?

**Comment by retioa11** [comment]
我直接转换yolov8n.pt到wts格式，再到engine格式时，遇到了类似的错误：
![image](https://github.com/wang-xinyu/tensorrtx/assets/81463532/1c3fbee8-6785-4e10-bdb8-bede9ae1e570)

**Comment by duongthanhdong** [comment]
I have saved the previous version that fixed with custom object model ( mentioned in #1325 ) and work fine for me (vehicles detection model) . If you'd like, I can share it with you. In addition to, altermative solutions are https://github.com/triple-Mu/YOLOv8-TensorRT.git or https://github.com/PaddlePaddle/FastDeploy/tree/develop/examples/vision/detection/yolov8/python 
.

> it seems that kNumClass in config.h didn't work?



> I meet exactly same error. /(ㄒoㄒ)/~~ ![image](https://user-images.githubusercontent.com/85165561/269517601-9527a3e2-1af1-4995-b97f-4b5dcfb6bace.png)



> ## Env
> * GPU, NVIDIA TITAN Xp
> * OS, Ubuntu20.04
> * Cuda version 11.3
> * TensorRT version 8.2.4
> 
> ## About this repo
> * which branch/tag/commit are you using?  [3cbb7cd](https://github.com/wang-xinyu/tensorrtx/commit/3cbb7cd25796509cb49cc2251020d589d98228c9)
> * which model?  yolov8
> 
> ## Your problem
> I trained yolov8 on Visdrone 2019. and generated engine file with gen_wts.py, and I changed kNumClass in config.h, when I ran ` ./yolov8 -s visdrone-8n-10c.wts visdrone-8n-10c.engine n` I got error output
> 
> ```
> Loading weights: /space/model/visdrone-8n-10c.wts
> [09/11/2023-23:44:03] [W] [TRT] The implicit batch dimension mode has been deprecated. Please create the network with NetworkDefinitionCreationFlag::kEXPLICIT_BATCH flag whenever possible.
> [09/11/2023-23:44:03] [E] [TRT] 3: (Unnamed Layer* 239) [Convolution]:kernel weights has count 36864 but 46080 was expected
> [09/11/2023-23:44:03] [E] [TRT] 4: (Unnamed Layer* 239) [Convolution]: count of 36864 weights in kernel, but kernel dimensions (3,3) with 64 input channels, 80 output channels and 1 groups were specified. Expected Weights count is 64 * 3*3 * 80 / 1 = 46080
> [09/11/2023-23:44:03] [E] [TRT] 4: [convolutionNode.cpp::computeOutputExtents::43] Error Code 4: Internal Error ((Unnamed Layer* 239) [Convolution]: number of kernel weights does not match tensor dimensions)
> [09/11/2023-23:44:03] [E] [TRT] 3: (Unnamed Layer* 239) [Convolution]:kernel weights has count 36864 but 46080 was expected
> [09/11/2023-23:44:03] [E] [TRT] 4: (Unnamed Layer* 239) [Convolution]: count of 36864 weights in kernel, but kernel dimensions (3,3) with 64 input channels, 80 output channels and 1 groups were specified. Expected Weights count is 64 * 3*3 * 80 / 1 = 46080
> [09/11/2023-23:44:03] [E] [TRT] 4: [convolutionNode.cpp::computeOutputExtents::43] Error Code 4: Internal Error ((Unnamed Layer* 239) [Convolution]: number of kernel weights does not match tensor dimensions)
> [09/11/2023-23:44:03] [E] [TRT] 3: (Unnamed Layer* 239) [Convolution]:kernel weights has count 36864 but 46080 was expected
> [09/11/2023-23:44:03] [E] [TRT] 4: (Unnamed Layer* 239) [Convolution]: count of 36864 weights in kernel, but kernel dimensions (3,3) with 64 input channels, 80 output channels and 1 groups were specified. Expected Weights count is 64 * 3*3 * 80 / 1 = 46080
> [09/11/2023-23:44:03] [E] [TRT] 4: [convolutionNode.cpp::computeOutputExtents::43] Error Code 4: Internal Error ((Unnamed Layer* 239) [Convolution]: number of kernel weights does not match tensor dimensions)
> [09/11/2023-23:44:03] [E] [TRT] 3: (Unnamed Layer* 239) [Convolution]:kernel weights has count 36864 but 46080 was expected
> [09/11/2023-23:44:03] [E] [TRT] 4: (Unnamed Layer* 239) [Convolution]: count of 36864 weights in kernel, but kernel dimensions (3,3) with 64 input channels, 80 output channels and 1 groups were specified. Expected Weights count is 64 * 3*3 * 80 / 1 = 46080
> [09/11/2023-23:44:03] [E] [TRT] 4: [convolutionNode.cpp::computeOutputExtents::43] Error Code 4: Internal Error ((Unnamed Layer* 239) [Convolution]: number of kernel weights does not match tensor dimensions)
> ```

**Comment by albert99zhy** [comment]
Have you solved it? I have a similar problem. I want to know how to deal with it, Please. 
![IMG_2164](https://github.com/wang-xinyu/tensorrtx/assets/126946332/4870c4a3-a1c8-4198-adf1-aa0ebe6ca45b)

**Comment by Xiaofeng-used** [comment]
> I have saved the previous version that fixed with custom object model ( mentioned in #1325 ) and work fine for me (vehicles detection model) . If you'd like, I can share it with you. In addition to, altermative solutions are https://github.com/triple-Mu/YOLOv8-TensorRT.git or https://github.com/PaddlePaddle/FastDeploy/tree/develop/examples/vision/detection/yolov8/python .
> 
> > it seems that kNumClass in config.h didn't work?
> 
> > I meet exactly same error. /(ㄒoㄒ)/~~ ![image](https://user-images.githubusercontent.com/85165561/269517601-9527a3e2-1af1-4995-b97f-4b5dcfb6bace.png)
> 
> > ## Env
> > 
> > * GPU, NVIDIA TITAN Xp
> > * OS, Ubuntu20.04
> > * Cuda version 11.3
> > * TensorRT version 8.2.4
> > 
> > ## About this repo
> > 
> > * which branch/tag/commit are you using?  [3cbb7cd](https://github.com/wang-xinyu/tensorrtx/commit/3cbb7cd25796509cb49cc2251020d589d98228c9)
> > * which model?  yolov8
> > 
> > ## Your problem
> > I trained yolov8 on Visdrone 2019. and generated engine file with gen_wts.py, and I changed kNumClass in config.h, when I ran ` ./yolov8 -s visdrone-8n-10c.wts visdrone-8n-10c.engine n` I got error output
> > ```
> > Loading weights: /space/model/visdrone-8n-10c.wts
> > [09/11/2023-23:44:03] [W] [TRT] The implicit batch dimension mode has been deprecated. Please create the network with NetworkDefinitionCreationFlag::kEXPLICIT_BATCH flag whenever possible.
> > [09/11/2023-23:44:03] [E] [TRT] 3: (Unnamed Layer* 239) [Convolution]:kernel weights has count 36864 but 46080 was expected
> > [09/11/2023-23:44:03] [E] [TRT] 4: (Unnamed Layer* 239) [Convolution]: count of 36864 weights in kernel, but kernel dimensions (3,3) with 64 input channels, 80 output channels and 1 groups were specified. Expected Weights count is 64 * 3*3 * 80 / 1 = 46080
> > [09/11/2023-23:44:03] [E] [TRT] 4: [convolutionNode.cpp::computeOutputExtents::43] Error Code 4: Internal Error ((Unnamed Layer* 239) [Convolution]: number of kernel weights does not match tensor dimensions)
> > [09/11/2023-23:44:03] [E] [TRT] 3: (Unnamed Layer* 239) [Convolution]:kernel weights has count 36864 but 46080 was expected
> > [09/11/2023-23:44:03] [E] [TRT] 4: (Unnamed Layer* 239) [Convolution]: count of 36864 weights in kernel, but kernel dimensions (3,3) with 64 input channels, 80 output channels and 1 groups were specified. Expected Weights count is 64 * 3*3 * 80 / 1 = 46080
> > [09/11/2023-23:44:03] [E] [TRT] 4: [convolutionNode.cpp::computeOutputExtents::43] Error Code 4: Internal Error ((Unnamed Layer* 239) [Convolution]: number of kernel weights does not match tensor dimensions)
> > [09/11/2023-23:44:03] [E] [TRT] 3: (Unnamed Layer* 239) [Convolution]:kernel weights has count 36864 but 46080 was expected
> > [09/11/2023-23:44:03] [E] [TRT] 4: (Unnamed Layer* 239) [Convolution]: count of 36864 weights in kernel, but kernel dimensions (3,3) with 64 input channels, 80 output channels and 1 groups were specified. Expected Weights count is 64 * 3*3 * 80 / 1 = 46080
> > [09/11/2023-23:44:03] [E] [TRT] 4: [convolutionNode.cpp::computeOutputExtents::43] Error Code 4: Internal Error ((Unnamed Layer* 239) [Convolution]: number of kernel weights does not match tensor dimensions)
> > [09/11/2023-23:44:03] [E] [TRT] 3: (Unnamed Layer* 239) [Convolution]:kernel weights has count 36864 but 46080 was expected
> > [09/11/2023-23:44:03] [E] [TRT] 4: (Unnamed Layer* 239) [Convolution]: count of 36864 weights in kernel, but kernel dimensions (3,3) with 64 input channels, 80 output channels and 1 groups were specified. Expected Weights count is 64 * 3*3 * 80 / 1 = 46080
> > [09/11/2023-23:44:03] [E] [TRT] 4: [convolutionNode.cpp::computeOutputExtents::43] Error Code 4: Internal Error ((Unnamed Layer* 239) [Convolution]: number of kernel weights does not match tensor dimensions)
> > ```

thank you sincerely, I successed testing yolov8_trt.py since lindsayshuo Fixed network issues with yolov8n #1365 .

**Comment by seawater668** [comment]
@wang-xinyu 跟官方的yolov8直接导出tensorRT有什么不一样吗，不知道官方的yolov8是怎么做到一键式操作的，他是经过onnx转的吗
